lib/config/plugins.js

```diff
@@ -105,18 +109,20 @@ module.exports = {
     load(pluginName) {
         const pluginNamespace = getNamespace(pluginName),
             pluginNameWithoutNamespace = removeNamespace(pluginName),
-            pluginNameWithoutPrefix = removePrefix(pluginNameWithoutNamespace);
+            pluginNameWithoutPrefix = removePrefix(pluginNameWithoutNamespace),
+            shortName = pluginNamespace + pluginNameWithoutPrefix,
+            longName = pluginNamespace + PLUGIN_NAME_PREFIX + pluginNameWithoutPrefix;
         let plugin = null;
 
-        if (!plugins[pluginNameWithoutPrefix]) {
+        if (!plugins[shortName]) {
             try {
-                plugin = require(pluginNamespace + PLUGIN_NAME_PREFIX + pluginNameWithoutPrefix);
+                plugin = require(longName);
             } catch (err) {
-                debug("Failed to load plugin eslint-plugin-" + pluginNameWithoutPrefix + ". Proceeding without it.");
+                debug("Failed to load plugin " + longName + ". Proceeding without it.");
```

Comment:
Unrelated to this PR, but I'm not sure "Proceeding without it." makes sense here since we're throwing an exception and killing the CLIEngine process. If you agree, I can make an issue about fixing that debug statement.